rosetta-lang/src/main/java/com/regnosys/rosetta/formatting2/RuneFormatter.java

```diff
@@ -0,0 +1,82 @@
+package com.regnosys.rosetta.formatting2;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.formatting2.FormatterRequest;
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionAccess;
+import org.eclipse.xtext.formatting2.regionaccess.ITextReplacement;
+import org.eclipse.xtext.formatting2.regionaccess.TextRegionAccessBuilder;
+import org.eclipse.xtext.formatting2.regionaccess.internal.TextRegionRewriter;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.util.TextRegion;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+public class RuneFormatter implements CodeFormatterService{
+	@Inject
+	private Provider<FormatterRequest> formatterRequestProvider;
+	
+	@Inject 
+	private Provider<RosettaFormatter> rosettaFormatterProvider;
+
+	@Inject
+	private TextRegionAccessBuilder regionBuilder;
+
+	@Override
+	public List<Resource> formatCollection(List<Resource> resources) throws IOException {
+		
+		List<Resource> result = new ArrayList<Resource>();
+		
+		for (Resource doc: resources) {
+			Resource formattedDocument = formatDocument(doc);
+			result.add(formattedDocument);
+		}	
+		return result;
+	}
+
+	private Resource formatDocument(Resource doc) throws IOException {
```

Comment:
Since the word `Document` is used for something specific in Xtext's formatting code (e.g., `IFormattableDocument`), the name `formatDocument` might be misleading. What about `formatResource`?

Reply:
I wonder if we should expose this method in the interface as well. It might be good if at some point a use case arises for formatting a single resource.